stock_reserve_area/models/stock_quant.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2023 ForgeFlow SL.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools import float_compare
+
+
+class StockQuant(models.Model):
+    _inherit = "stock.quant"
+
+    reserve_area_id = fields.Many2one(
+        "stock.reserve.area", compute="_compute_reserve_area_id", store=True
+    )
+
+    @api.depends("location_id")
+    def _compute_reserve_area_id(self):
+        for quant in self:
+            reserve_areas = self.env["stock.reserve.area"].search([])
+            for reserve_area in reserve_areas:
+                if reserve_area.is_location_in_area(quant.location_id):
+                    quant.reserve_area_id = reserve_area
+                    break
+
+    def _get_reserve_area_available_quantity(self, product_id, reserve_area_id):
+        self = self.sudo()
+        quants = self._gather_reserve_area(product_id, reserve_area_id)
+        rounding = product_id.uom_id.rounding
+        # for the moment not taking into account lots
+        # if product_id.tracking == 'none':
+        area_reserved_quantity = (
+            self.env["stock.move"]
+            .search(
+                [
+                    ("reserve_area_id", "=", reserve_area_id.id),
+                    ("product_id", "=", product_id.id),
+                ]
+            )
+            .mapped("area_reserved_availability")
+        )
+        available_quantity = sum(quants.mapped("quantity")) - sum(
+            area_reserved_quantity
+        )
```

Comment:
Pay attention that if you have lots of quants and/or lots of moves, that a lot of data to fetch to compute the sum. Maybe at some point, you'll want to replace it by read_group